framework/web/Cookie.php

```diff
@@ -65,4 +67,38 @@ public function __toString()
     {
         return (string) $this->value;
     }
+
+    /**
+     * Unserializes a cookie received by the client.
+     * @param string $data json encoded cookie value
+     * @return self|null
+     * @since 2.0.12
+     */
+    public static function fromDataString($data)
+    {
+        if (($unserialized = json_decode($data, true)) !== false) {
```

Comment:
What should the behavior be regarding errors?
If I pass an invalid cookie, should the request object just crash (ie bubble up the exception) or should it silently ignore the cookie?